src/Bicep.Cli/Commands/FormatCommand.cs

```diff
@@ -54,50 +27,56 @@ public int Run(FormatArguments args)
+        var context = PrettyPrinterV2Context.Create(options, parser.LexingErrorLookup, parser.ParsingErrorLookup);
 
-        if (featureProvider.PrettyPrintingEnabled)
-        {
-            var v2Options = this.configurationManager.GetConfiguration(inputUri).Formatting.Data;
-            var printerV2Context = PrettyPrinterV2Context.Create(v2Options, parser.LexingErrorLookup, parser.ParsingErrorLookup);
-
-            if (args.OutputToStdOut)
-            {
-                PrettyPrinterV2.PrintTo(io.Output, program, printerV2Context);
-                io.Output.Flush();
-            }
-            else
-            {
-                var outputPath = PathHelper.ResolveDefaultOutputPath(inputUri.LocalPath, args.OutputDir, args.OutputFile, path => path);
-                using var writer = new StreamWriter(outputPath);
-
-                PrettyPrinterV2.PrintTo(writer, program, printerV2Context);
-            }
-
-            return 0;
-        }
-
-        var options = new PrettyPrintOptions(
-            args.Newline ?? NewlineOption.Auto,
-            args.IndentKind ?? IndentKindOption.Space,
-            args.IndentSize ?? 2,
-            args.InsertFinalNewline ?? false
-        );
-
-        var output = PrettyPrinter.PrintProgram(program, options, parser.LexingErrorLookup, parser.ParsingErrorLookup);
         if (args.OutputToStdOut)
         {
-            io.Output.Write(output);
+            PrettyPrinterV2.PrintTo(io.Output, program, context);
             io.Output.Flush();
         }
         else
         {
-            static string DefaultOutputPath(string path) => path;
-            var outputPath = PathHelper.ResolveDefaultOutputPath(inputUri.LocalPath, args.OutputDir, args.OutputFile, DefaultOutputPath);
+            var outputPath = PathHelper.ResolveDefaultOutputPath(inputUri.LocalPath, args.OutputDir, args.OutputFile, path => path, fileSystem);
+            using var fileStream = fileSystem.File.Open(outputPath, FileMode.Create, FileAccess.Write, FileShare.Read);
+            using var writer = new StreamWriter(fileStream);
 
-            File.WriteAllText(outputPath, output);
+
+            PrettyPrinterV2.PrintTo(writer, program, context);
         }
 
         return 0;
     }
+
+    private PrettyPrinterV2Options GetPrettyPrinterOptions(Uri inputUri, FormatArguments args)
+    {
+        var options = configurationManager.GetConfiguration(inputUri).Formatting.Data;
+
+        if (args.NewlineKind is not null)
+        {
+            options = options with { NewlineKind = args.NewlineKind.Value };
+        }
+
+        if (args.IndentKind is not null)
+        {
+            options = options with { IndentKind = args.IndentKind.Value };
+        }
+
+        if (args.IndentSize is not null)
+        {
+            options = options with { IndentSize = args.IndentSize.Value };
+        }
+
+        if (args.Width is not null)
+        {
+            options = options with { Width = args.Width.Value };
```

Comment:
I'm slightly concerned that people may be opinionated about making the decision to automatically wrap vs preserving formatting.

For example, if I've chosen to write:
```bicep
var foo = [
           'short'
]
```

I might expect the formatter to format it to:
```bicep
var foo = [
  'short'
]
```

But this feels unexpected and possibly undesirable:
```bicep
var foo = [ 'short' ]
```

Is there any ability to configure this behavior?

Reply:
Currently, no, but this is implemented for objects, and we can easily turn it on for arrays. Would you consider this a valuable topic to bring up for discussion at our upcoming Bicep community call?

Reply:
By the way, my point here isn't to say "my way is better", I'm just using myself as an example of "people can be have strong opinions about formatting".

Reply:
I think this may make more sense if we end up deciding to use a larger `width` (say `120`) as it may lead to long single line arrays 🤔.

Reply:
> Currently, no, but this is implemented for objects, and we can easily turn it on for arrays. Would you consider this a valuable topic to bring up for discussion at our upcoming Bicep community call?

Yeah, I think it definitely wouldn't hurt to bring up. I'm not sure what sort of usage the current formatter gets, so it's difficult to gauge how important it is.

Reply:
We could also introduce a setting to allow users to refine this behavior further. A useful feature might be the ability to split objects and arrays not just based on width, but also considering the quantity of properties or items they contain (e.g., always break an array if it contains more than 2 items).